### PR TITLE
Add test coverage for clean()

### LINKED ISSUE #256: Add test coverage for `clean()`
`clean()` currently has no dedicated tests, which allowed #249 to go undetected.

The function handles several distinct code paths that should be tested:

1. Country name normalisation - 2-letter ISO, 3-letter ISO, and full country names
2. Non-numeric age entries - entries without numbers should become NA
3. "Under X" format - e.g., "Under 1" → "0-1"
4. Age unit conversion - months/weeks to years
5. Age range parsing - "20-30" → part_age_est_min = 20, part_age_est_max = 30
6. Exact age column creation - part_age_exact from numeric ages

A regression test for (5) should be added to #250, but more comprehensive coverage would help prevent future regressions.

### PR DESCRIPTION
**Pull request by sbfnk**
## Summary

- Add comprehensive test coverage for `clean()` function (42 tests)
- Fix multiple bugs in `clean()`:
  - Data.table syntax error (missing comma) that caused errors when processing age values with spaces
  - Age unit extraction not updating the age column to just the numeric part
  - `period()` function not being applied row-by-row for unit conversion
- Tests cover: country name normalisation, non-numeric age handling, "Under X" format conversion, age range parsing, age unit conversion (months/weeks to years), exact age column creation, and edge cases

Closes #256

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed age parsing and conversion for values with units (e.g., "6 months", "52 weeks").

* **Deprecations**
  * Public argument names moved to snake_case (e.g., age_limits); dot-named arguments still work but emit deprecation warnings.

* **Documentation & Examples**
  * Vignettes and examples updated to use new parameter names.

* **Tests**
  * Expanded tests for age parsing, country normalization, deprecated-argument warnings, CI skips, and related edge cases.

* **Chores**
  * Minor housekeeping (global variable tracking).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: R/pop_age.R [hunk 1]
```diff
@@ -16,41 +18,69 @@
...
-    missing.ages <- setdiff(
-      age.limits[age.limits <= max.age],
-      pop[[pop.age.column]]
+  if (!missing(age_limits)) {
+    age_limits <- sort(age_limits)
+    max_age <- max(pop[, pop_age_column, with = FALSE])
+    missing_ages <- setdiff(
+      age_limits[age_limits <= max_age],
+      pop[[pop_age_column]]
     )
```
[the remark this answers is not in the document]
Comment (sbfnk) [reply R/pop_age.R]: https://github.com/epiforecasts/socialmixr/issues/267

File: tests/testthat/test-clean.R [hunk 2]
```diff
@@ -1,8 +1,120 @@
...
+    part_id = 1:3,
+    part_age = c("Under 1", "Under 5", "Under 18"),
+    stringsAsFactors = FALSE
+  )
+  survey <- create_test_survey(participants)
+  cleaned <- clean(survey)
+
+  expect_identical(cleaned$participants$part_age_est_min, c(0, 0, 0))
+  expect_identical(cleaned$participants$part_age_est_max, c(1, 5, 18))
+})
```
[the remark this answers is not in the document]
Comment (sbfnk) [reply tests/testthat/test-clean.R]: lintr prefers `expect_identical()` and we'd want to catch unexpected type changes.
